tracing-subscriber/src/layer.rs

```diff
@@ -305,6 +305,14 @@ where
         let _ = (attrs, id, ctx);
     }
 
+    // TODO(eliza): do we want this to be a public API? If we end up moving
+    // filtering layers to a separate trait, we may no longer want `Layer`s to
+    // be able to participate in max level hinting...
+    #[doc(hidden)]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        None
+    }
```

Comment:
@davidbarsky @LucioFranco opinions?

Reply:
While we can break tracing-subscriber more readily, I think this should be a private/doc-hidden API for now and expose it via the Filter API.

Reply:
Yeah. I might feel otherwise if there were a lot of filter implementations other than the existing `EnvFilter` and `LevelFilter`, but in practice, I pretty much only ever see people use `EnvFilter`...

Reply:
Agreed, only if we really need it to be public.